### PR TITLE
segger: rm spin_lock_irqsave(NULL) in drivers/segger/config/SEGGER_RTT_Conf.h

### PR DESCRIPTION
**Pull request by hujun260**
## Summary
segger: rm spin_lock_irqsave(NULL) in drivers/segger/config/SEGGER_RTT_Conf.h

## Impact
segger
img_v3_02ic_b58e3a17-88ed-4b8d-9d5a-a2673a3d405l

## Testing
ci

### CODE REVIEW

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



No, this PR does not fully meet the NuttX requirements as described.  While it provides a summary of the *what*, it lacks crucial information regarding the *why*.

Here's a breakdown of what's missing:

* **Summary:**
    * **Missing:**  The PR summary needs to explain *why* `spin_lock_irqsave(NULL)` is being removed. What problem does this fix? What were the negative consequences of having it there?  What was the original intent of that code, and why is it no longer necessary or correct?
    * **Missing:**  What functional part of the code is being changed?  It mentions `drivers/segger/config/SEGGER_RTT_Conf.h`, but what does that file control? What aspects of the Segger functionality are affected?
    * **Missing:** How does the removal of `spin_lock_irqsave(NULL)` work? What mechanisms replace it, if any?
    * **Missing:** Links to related NuttX or NuttX Apps issues or pull requests.  Even if there isn't a directly related issue, it's helpful to link to discussions or documentation that provides context.

* **Impact:**
    * **Insufficient:**  The image provided doesn't give enough context.  What does "segger" refer to in the image? Is this a build log excerpt? An error message?  It's not clear what's being shown.  Explicitly answer all the impact questions (user, build, hardware, documentation, security, compatibility) with "NO" or "YES" and provide detailed explanations where necessary.
    * **Missing:** Explicit "NO" or "YES" answers to *all* impact questions.  Just saying "segger" and providing an image isn't sufficient.

* **Testing:**
    * **Insufficient:**  "ci" isn't enough information.  Specify the CI environment. Which build hosts and targets were tested? What *specific tests* were run?  The CI logs themselves should be linked, not just a mention of "ci."
    * **Missing:**  "Testing logs before change" and "Testing logs after change" are completely empty. This is a critical part of demonstrating that the change has the intended effect and doesn't introduce regressions.  Even if the change seems small, demonstrable proof is required.


In short, the PR needs significantly more detail and context to meet the NuttX requirements. It needs to clearly articulate the rationale behind the change, its effects, and provide verifiable evidence that it works as intended.

File: drivers/segger/config/SEGGER_RTT_Conf.h [hunk 1]
```diff
@@ -81,11 +89,11 @@
 
 /* Lock RTT (nestable)   (i.e. disable interrupts) */
 
-#define SEGGER_RTT_LOCK()               irqstate_t __flags = spin_lock_irqsave(NULL)
+#define SEGGER_RTT_LOCK()               irqstate_t __flags = spin_lock_irqsave(&g_segger_lock)
```
Comment (anchao) [inline drivers/segger/config/SEGGER_RTT_Conf.h:92]:
```suggestion
#define SEGGER_RTT_LOCK()               irqstate_t __flags = spin_lock_irqsave_wo_note(&g_segger_lock)
```

Comment (hujun260) [reply drivers/segger/config/SEGGER_RTT_Conf.h]: It looks fine！
img_v3_02ic_8645028e-1d37-4122-9c7f-65f0297c2cbl

File: drivers/segger/note_sysview.c [hunk 2]
```diff
@@ -162,6 +162,12 @@ static struct note_sysview_driver_s g_note_sysview_driver =
  * Private Functions
  ****************************************************************************/
 
+/****************************************************************************
+ * Public Data
+ ****************************************************************************/
+
+spinlock_t g_segger_lock = SP_UNLOCKED;
```
Comment (anchao) [inline drivers/segger/note_sysview.c:169]: move before "Private Functions"

**Review by anchao — APPROVED** [review]
